components/core/tests/test-end_to_end.cpp

```diff
@@ -0,0 +1,127 @@
+#include <sys/wait.h>
+
+#include <cstdlib>
+#include <filesystem>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+#include <fmt/format.h>
+
+#include "../src/clp_s/JsonConstructor.hpp"
+#include "../src/clp_s/JsonParser.hpp"
+
+constexpr std::string_view cTestEndToEndArchiveDirectory{"test-end-to-end-archive"};
+constexpr std::string_view cTestEndToEndOutputDirectory{"test-end-to-end-out"};
+constexpr std::string_view cTestEndToEndOutputSortedJson{"test-end-to-end_sorted.json"};
+constexpr std::string_view cTestEndToEndInputFileDirectory{"test_log_files"};
+constexpr std::string_view cTestEndToEndInputFile{"test_no_floats_sorted.json"};
+
+namespace {
+/**
+ * Class with no data members who's contruction and destruction is soley
+ *      for cleanup up files and directorys created by the test case.
+ */
+class Cleanup {
+public:
+    Cleanup() { delete_files(); }
+
+    ~Cleanup() { delete_files(); }
+
+private:
+    static void delete_files() {
+        std::filesystem::remove_all(cTestEndToEndArchiveDirectory);
+        std::filesystem::remove_all(cTestEndToEndOutputDirectory);
+        std::filesystem::remove(cTestEndToEndOutputSortedJson);
+    }
+};
+
+auto get_test_input_path_relative_to_tests_dir() -> std::filesystem::path {
+    return std::filesystem::path{cTestEndToEndInputFileDirectory} / cTestEndToEndInputFile;
+}
+
+auto get_test_input_local_path() -> std::string {
+    std::filesystem::path const current_file_path{__FILE__};
+    auto const tests_dir{current_file_path.parent_path()};
+    return (tests_dir / get_test_input_path_relative_to_tests_dir()).string();
+}
+}  // namespace
+
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+TEST_CASE("clp-s_compression_and_extraction_no_floats", "[clp-s][end-to-end]") {
```

Comment:
```suggestion
TEST_CASE("clp-s-compress-extract-no-floats", "[clp-s][end-to-end]") {
```

I think that's more compact but with the same meaning?